modules/dasBidAdapter.js

```diff
@@ -0,0 +1,240 @@
+import { getAllOrtbKeywords } from '../libraries/keywords/keywords.js';
+import { getAdUnitSizes } from '../libraries/sizeUtils/sizeUtils.js';
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import { BANNER } from '../src/mediaTypes.js';
+import { deepAccess } from '../src/utils.js';
+
+const BIDDER_CODE = 'das';
+
+const getEndpoint = (network) => {
+  return `https://csr.onet.pl/${encodeURIComponent(network)}/bid`;
+};
+
+function parseParams(params, bidderRequest) {
+  const customParams = {};
+  const keyValues = {};
+
+  if (params.site) {
+    customParams.site = params.site;
+  }
+
+  if (params.area) {
+    customParams.area = params.area;
+  }
+
+  if (params.network) {
+    customParams.network = params.network;
+  }
+
+  // Custom parameters
+  if (params.customParams && typeof params.customParams === 'object') {
+    Object.assign(customParams, params.customParams);
+  }
+
+  const pageContext = params.pageContext;
+  if (pageContext) {
+    // Document URL override
+    if (pageContext.du) {
+      customParams.du = pageContext.du;
+    }
+
+    // Referrer override
+    if (pageContext.dr) {
+      customParams.dr = pageContext.dr;
+    }
+
+    // Document version
+    if (pageContext.dv) {
+      customParams.DV = pageContext.dv;
+    }
+
+    // Keywords
+    const keywords = getAllOrtbKeywords(
+      bidderRequest?.ortb2,
+      pageContext.keyWords,
+    );
+    if (keywords.length > 0) {
+      customParams.kwrd = keywords.join('+');
+    }
+
+    // Local capping
+    if (pageContext.capping) {
+      customParams.local_capping = pageContext.capping;
+    }
+
+    // Key values
+    if (pageContext.keyValues && typeof pageContext.keyValues === 'object') {
+      Object.entries(pageContext.keyValues).forEach(([key, value]) => {
+        keyValues[`kv${key}`] = value;
+      });
+    }
+  }
+
+  const du = customParams.du || deepAccess(bidderRequest, 'refererInfo.page');
+  const dr = customParams.dr || deepAccess(bidderRequest, 'refererInfo.ref');
+
+  if (du) customParams.du = du;
+  if (dr) customParams.dr = dr;
+
+  const dsaRequired = deepAccess(bidderRequest, 'ortb2.regs.ext.dsa.required');
+  if (dsaRequired !== undefined) {
+    customParams.dsainfo = dsaRequired;
+  }
+
+  return {
+    customParams,
+    keyValues,
+  };
+}
+
+function buildUserIds(customParams) {
+  const userIds = {};
+  if (customParams.lu) {
+    userIds.lu = customParams.lu;
+  }
+  if (customParams.aid) {
+    userIds.aid = customParams.aid;
+  }
+  return userIds;
+}
+
+function getNpaFromPubConsent(pubConsent) {
+  const params = new URLSearchParams(pubConsent);
+  return params.get('npa') == '1';
+}
+
+function buildOpenRTBRequest(bidRequests, bidderRequest) {
+  const { customParams, keyValues } = parseParams(
+    bidRequests[0].params,
```

Comment:
W obecnym adapterze jest to zrobione tak:
```js
const bid = [...bidRequests].reverse().find(bid => bid.params?.customParams?.bidPriority === 1) || bidRequests[0];
```
Z tego co patrzyłem w dlApi, jest to potrzebne, żeby działały bety z Prebida. Chyba nie możemy bez tego wdrożyć nowego adaptera, bo AdOpsi zaczęli już z tego korzystać.

Reply:
Bety na ten moment chciałem odpuścić bo to będzie zupełnie inaczej zaimplementowane niż jak jest do tej pory. Temat pewnie będzie do zaadresowania w Q2

Reply:
OK, ale w takim razie, jak będziemy to wdrażać, żeby nie popsuć tej funkcjonalności AdOpsom? Na razie tylko do wewnętrznych testów, bez włączania do produkcyjnego ruchu?

Reply:
bety cały czas będą działać starą ścieżką. Przed całkowitym wyłączeniem starej ścieżki dorobimy ten ficzer